runtime/caml/io.h

```diff
@@ -42,9 +42,7 @@ struct channel {
   char * max;                   /* Logical end of the buffer (for input) */
   void * mutex;                 /* Placeholder for mutex (for systhreads) */
   struct channel * next, * prev;/* Double chaining of channels (flush_all) */
-  int revealed;                 /* For Cash only */
-  int old_revealed;             /* For Cash only */
-  int refcount;                 /* For flush_all and for Cash */
+  int refcount;                 /* For flush_all */
```

Comment:
`refcount` can go too. It is only increased inside `caml_alloc_channel` to prevent a race with finalisation (#8191), but finalisation no longer happens during allocation from C.

Reply:
Upon closer inspection `refcount` needs to stay, but the comment `/* prevent finalization during next alloc */` should be fixed to better match the purpose of the reference count :)

The test for `refcount > 0` inside `caml_close_channel` is now pointless.

Reply:
I would document
```ocaml
  int refcount;                 /* Number of custom blocks owning the struct */
```
and remove `CHANNEL_FLAG_MANAGED_BY_GC`.